sklearn/preprocessing/imputation.py

```diff
@@ -189,6 +194,14 @@ def _sparse_fit(self, X, strategy, missing_values, axis):
         else:
             X = X.tocsc()
 
+        if axis == 0:
+            mask = _get_mask(X.data, self.missing_values)
+            mask_matrix = X.__class__((mask, X.indices.copy(),
+                                       X.indptr.copy()), shape=X.shape,
+                                      dtype=X.dtype)
+            mask_matrix.eliminate_zeros()  # removes explicit False entries
```

Comment:
If we're calling `sum` (and not doing anything else with `mask_matrix`), we don't get any benefit from this line. Remove it.